sdk/search/ai-search/README.md

```diff
@@ -0,0 +1,302 @@
+
+## Getting started
+
+### Currently supported environments
+
+- Node.js version 8.x.x or higher
+
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+- An existing [Cognitive Search][search_resource] resource. If you need to create the resource, you can use the [Azure Portal][azure_portal] or [Azure CLI][azure_cli].
+
+If you use the Azure CLI, replace `<your-resource-group-name>` and `<your-resource-name>` with your own unique names:
+
+```PowerShell
+az search service create --resource-group <your-resource-group-name> --name <your-resource-name> --sku S
+```
+
+The above creates a resource with the "Standard" pricing tier. See [choosing a pricing tier](https://docs.microsoft.com/en-us/azure/search/search-sku-tier) for more information.
+
+### 1. Install the `@azure/ai-search` package
+
+```bash
+npm install @azure/ai-search
+```
+
+### 2. Create and authenticate a `SearchIndexClient`
+
+Cognitive Search uses keys for authentication.
+
+#### Using an Admin Key
+
+Use the [Azure CLI][azure_cli] snippet below to get the Admin Key from the Cognitive Search resource.
+
+```PowerShell
+az search admin-key show --resource-group <your-resource-group-name> --service-name <your-resource-name>
+```
+
+Alternatively, you can get the endpoint and Admin Key from the resource information in the [Azure Portal][azure_portal].
+
+Once you have an Admin Key, you can use it as follows:
+
+```js
+const { SearchIndexClient, SearchApiKeyCredential } = require("@azure/ai-search");
+
+const client = new SearchIndexClient(
+  "<endpoint>",
+  "<indexName>",
+  new SearchApiKeyCredential("<Admin Key>");
+);
+```
+
+## Key concepts
+
+### SearchIndexClient
+
+`SearchIndexClient` is one of the primary interface for developers using the Cognitive Search client library. It provides asynchronous methods for working with documents in an index.
+
+## Examples
+
+### Query documents in an index
+
+To list all results of a particular query, you can use `listSearchResults` with a search string that uses [simple query syntax](https://docs.microsoft.com/en-us/azure/search/query-simple-syntax):
+
+```js
+for await (const result of client.listSearchResults("wifi -luxury")) {
+  console.log(result);
+}
+```
+
+For a more advanced search that uses [Lucene syntax](https://docs.microsoft.com/en-us/azure/search/query-lucene-syntax), specify `queryType` to be `all`:
+
+```js
+for await (const result of client.listSearchResults('category:budget AND "recently renovated"^3', {
+  queryType: "full",
+  searchMode: "all"
+})) {
+  console.log(result);
+}
+```
+
+### Querying with OData filters
+
+Using the `filter` query parameter allows you to query an index using the syntax of an [OData \$filter expression](https://docs.microsoft.com/en-us/azure/search/search-query-odata-filter).
+
+```js
+const { odata } = require("@azure/ai-search");
+
+let baseRateMax = 200;
+let ratingMin = 4;
+for await (const result of client.listSearchResults("WiFi", {
```

Comment:
I seem to recall @willmtemple going through and making the readme snippets work as standalone code. 

So the `client` instantiation should be in here as well.

Funnily enough one of my snippets in appconfig has the full deal and one doesn't. So perhaps the "rule" is not so cut and dry...

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/appconfiguration/app-configuration/README.md#create-and-get-a-setting

Reply:
Hmm storage-blob does this too... but boy it sure feels heavy

Reply:
will do this for now